tripal/tests/src/Kernel/Services/TripalFileRetriever/TripalFileRetrieverTest.php

```diff
@@ -211,17 +216,32 @@ public function testTripalFileRetriever(string $url, array $options, array $expe
         'Did not receive NULL when we expect file retrieval to have failed.');
     }
 
-    // Tests downloadFile()
+    // Provide some non-default retrieval options.
+    // We can't easily test the retry option because that requires an
+    // intermittent internet connection.
+    $retrieval_options = [];
+    if ($expectations['test_rate_limit']) {
+      $retrieval_options = [
+        'rate_limit' => 8.765,
+        'retry_delay' => 2.345,
+      ];
+    }
+
+    // Tests downloadFile(), and additionally the rate-limiting parameter.
+    // The test above will have set the internal last_request_time.
```

Comment:
I think this makes it more clear how last_request_time is being set for those of us who don't know the file retrieval service inside and out.
```suggestion
    // Calling retrieveFileContents() above set the internal last_request_time.
```